File: 0101-0200/0136.Single_Number.cpp
//14ms




class Solution {
public:
    int singleNumber(vector<int>& nums) { 
       unordered_map<int,int> a;
	   for(auto x: nums)
		   a[x]++;
	   for(auto z:a)
		   if(z.second==1)
			   return z.first;
	   return -1;
    }
};




//7ms





class Solution {
public:
    int singleNumber(vector<int>& nums) { 
       sort(nums.begin(),nums.end());
        for(int i=1;i<nums.size();i+=2)
        {
            if(nums[i]!=nums[i-1])
                return nums[i-1];
        }
        return nums[nums.size()-1];
    }
};





//0ms





class Solution {
public:
    int singleNumber(vector<int>& nums) { 
       int ans=0;
	   for(auto x:nums)
	   ans^=x;
	   return ans;
    }
};





//0ms






class Solution {
public:
    int singleNumber(vector<int>& nums) {
        int x = 0;
        for(int i : nums)
            x ^= i;
        return x;
    }
};





//0ms





class Solution
{
public:
    int singleNumber(vector<int> &nums)
    {
        return accumulate(nums.begin(), nums.end(), 0, bit_xor<int>());
    }
};

File: 0101-0200/0136.Single_Number.rs
//0ms




use std::ops::BitXor;

impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        nums
            .into_iter()
            .fold(0, i32::bitxor)
    }
}




//0ms





impl Solution {
    pub fn single_number(nums: Vec<i32>) -> i32 {
        let mut xor=0;for i in 0..nums.len(){xor^=nums[i];}xor
    }
}


